src/files/filesProcessor.ts

```diff
@@ -0,0 +1,162 @@
+import { EgoJwtManager } from "auth";
+import { getFilesByProgramId } from "files/getFilesByProgramId";
+import _ from "lodash";
+import logger from "logger";
+import { initializeRdpcInfo } from "rdpc/analysesProcessor";
+import { DonorInfoMap, StringMap } from "rdpc/types";
+import {
+  DonorMolecularDataReleaseStatus,
+  File,
+  FileReleaseState,
+} from "./types";
+
+/** Gets file data from file-service, group files by donorId, and determine
+ *  donor's file release status based on the number of file status.
+ */
+export const determineReleaseStatus = async (
+  programId: string,
+  egoJwtManager: EgoJwtManager,
+  filesFetcher: typeof getFilesByProgramId,
+  donorIds?: string[]
+): Promise<DonorInfoMap> => {
+  const result: DonorInfoMap = {};
+  // gets files grouped by donor id
+  const files = await getFilesByPage({
+    programId: programId,
+    egoJwtManager: egoJwtManager,
+    filesFetcher: filesFetcher,
+    donorIds: donorIds,
+  });
+
+  Object.entries(files).forEach(([donorId, files]) => {
+    const filesByReleaseState = _.groupBy(files, (file) => file.releaseState);
+    const releaseStates = Object.keys(filesByReleaseState);
+
+    initializeRdpcInfo(result, donorId);
+
+    switch (releaseStates.length) {
+      case 1:
```

Comment:
Should you handle a case 0? would only happen if theres bad data in files service... maybe put a check for this and set the status to NO_RELEASE